src/service/viewer-impl.js

```diff
@@ -1012,8 +1012,13 @@ export class Viewer {
           eventType, data, awaitResponse));
     }
 
+    // If there is no viewer, no further action is needed.
     if (!this.messagingReadyPromise_) {
-      return Promise.reject(getChannelError());
+      if (awaitResponse) {
+        return Promise.reject(getChannelError());
+      } else {
+        return Promise.resolve();
```

Comment:
Actually, this is not equivalent to the code we had before https://github.com/ampproject/amphtml/pull/6508. To get back to that, we need something like:

```js
if (awaitPromise && !this.messagingReadyPromise_) {
  return Promise.reject(getChannelError());
}
```